### DISCUSSION TITLE
Syncing is not available behind an authenticated proxy

### DISCUSSION THREAD

**Discussion by gizmometer** [standardnotes/forum#2551] [General]
I am unable to reach the sync servers because I work behind an authenticated proxy.  Please consider adding support for this type of environment.

**Comment by atmoio** [comment]
Hey Conrad, does this only happen via the desktop app, or the web app as well?

**Comment by gizmometer** [comment]
Hi Mo, it's only on the desktop app.

**Comment by atmoio** [comment]
This issue has come up previously once in the past, and unfortunately the solution requires a lot of manual intervention in the code. I can't really promise when I can get to this, but I'll keep a close eye on it.

**Comment by dreamflasher** [comment]
I am running into this problem too, is there now any way to use the desktop app behind a proxy?

**Comment by atmoio** [comment]
Not that I know of currently..

**Comment by zicklag** [comment]
I'm running into this, too. I once added proxy support for the Astro CLI [here](https://github.com/withastro/astro/pull/4676), using `global-agent`, and it turned out the only code change was a few lines of initialization code:

```js
import {
	bootstrap as bootstrapGlobalAgent
} from 'global-agent';

bootstrapGlobalAgent({
	environmentVariableNamespace: '',
});
```

That should set it up to read the `http_proxy` environment variables.

I just found Standard Notes, so I'm not sure if I'll end up using it yet, but if it seems nice I could try a PR real quick to see if it actually ends up that easy to fix.

**Comment by effieeee** [comment]
Hi @zicklag, the app repo is here if you’d like to try submitting a PR! 
https://github.com/standardnotes/app/pulls

**Comment by bartlomiejzuber** [comment]
Hi, this is unfortunately again an issue with the newest version of astro 4.5.x.

Per my observations it looks like:
1. If we use `global-agent` to bootstrap on the frontmatter part then it works fine for that part
2. But it fails when astro tries to pull down the remote images via <Image /> component

It also seems that code from related MR has been removed on some later versions as I can't find it anywhere in the code.

**Comment by luca-salga** [comment]
Hi guys. I've landed here because I've just begun trialling StandardNotes with self-hosted sync. The sync server sits behind [Pangolin](https://pangolin.net). In order to fully benefit from the Pangolin access control and authentication layer, Standard Notes Desktop/mobile applications would need to implement Custom Proxy Headers, which is one of the forms of proxy authentication that Pangolin supports.

I've seen other applications that support this, such as Immich (photos) and Obsidian (community plugin for self-hosted sync).

Any chance Standard Notes will implement this in the future?
